superset/connectors/sqla/api.py

```diff
@@ -0,0 +1,812 @@
+          - in: path
+            schema:
+              type: integer
+            name: pk
+            description: The dataset id
+          - in: path
+            schema:
+              type: integer
+            name: column_id
+            description: The column id
+          requestBody:
+            description: Model schema
+            required: true
+            content:
+              application/json:
+                schema:
+                  $ref: '#/components/schemas/{{self.__class__.__name__}}.put'
+          responses:
+            200:
+              description: Item changed
+              content:
+                application/json:
+                  schema:
+                    type: object
+                    properties:
+                      result:
+                        $ref: '#/components/schemas/{{self.__class__.__name__}}.put'
+            400:
+              $ref: '#/components/responses/400'
+            401:
+              $ref: '#/components/responses/401'
+            403:
+              $ref: '#/components/responses/403'
+            404:
+              $ref: '#/components/responses/404'
+            422:
+              $ref: '#/components/responses/422'
+            500:
+              $ref: '#/components/responses/500'
+        """
+        request.json["table"] = item.id
+        return self.put_headless(column_id)
+
+    @expose("/<int:pk>/column/<column_id>", methods=["DELETE"])
+    @protect()
+    @check_ownership_dataset_exists
+    @safe
+    def delete(  # pylint: disable=arguments-differ
+        self, item: SqlaTable, column_id: int  # pylint: disable=unused-argument
+    ) -> Response:
+        """Delete a column from a dataset
+        ---
+        delete:
+          description: >-
+            Delete a column from a dataset
+          parameters:
+          - in: path
+            schema:
+              type: integer
+            name: pk
+          responses:
+            200:
+              description: Item deleted
+              content:
+                application/json:
+                  schema:
+                    type: object
+                    properties:
+                      message:
+                        type: string
+            403:
+              $ref: '#/components/responses/403'
+            404:
+              $ref: '#/components/responses/404'
+            422:
+              $ref: '#/components/responses/422'
+            500:
+              $ref: '#/components/responses/500'
+        """
+        return self.delete_headless(column_id)
+
+
+class DatasetMetricRestApi(BaseSupersetModelRestApi):
+    datamodel = SQLAInterface(SqlMetric)
+
+    resource_name = "dataset"
+    allow_browser_login = True
+    class_permission_name = "SqlMetricInlineView"
+    include_route_methods = RouteMethod.REST_MODEL_VIEW_CRUD_SET | {RouteMethod.RELATED}
+    openapi_spec_tag = "Datasets"
+
+    list_columns = ["metric_name", "verbose_name", "metric_type"]
```

Comment:
Might be overkill (some metrics might be very long) and probably depends on where this is used in practice, but for a listing I think I would prefer to see the `expression` over `metric_type` here. Looking at the datasource editor, it seems it's also doing the same:

![image](https://user-images.githubusercontent.com/33317356/74379795-16036600-4df1-11ea-9387-56d45e9fff9c.png)